firebase-vscode/src/analytics.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  env,
+  TelemetryLogger,
+  TelemetrySender,
+} from "vscode";
+import { pluginLogger } from "./logger-wrapper";
+import { AnalyticsParams, trackVSCode } from "./track";
+
+
+export class AnalyticsLogger {
+    readonly logger: TelemetryLogger
+    constructor() {
+        this.logger = env.createTelemetryLogger(new GA4TelemetrySender(pluginLogger));
+    }
+}
+
+class GA4TelemetrySender implements TelemetrySender {
+  constructor(readonly pluginLogger) {}
+
+  sendEventData(
+    eventName: string,
+    data?: Record<string, any> | undefined,
+  ): void {
+    if (!env.isTelemetryEnabled) {
+      pluginLogger.warn("Telemetry is not enabled.")
```

Comment:
We should return here, otherwise we just log that telemetry is disabled and then send telemetry 😨

Reply:
I believe the telemetrylogger from vscode handles this. Let me confirm

Reply:
never mind.  we should definitely return